byteps/common/cpu_reducer.cc

```diff
@@ -56,171 +56,199 @@ bool CpuReducer::isRoot() {
 }
 #endif
 
-int CpuReducer::sum(void* dst, void* src, size_t len, DataType dtype) {
+int CpuReducer::sum(void* dst, const void* src, size_t len, DataType dtype,
```

Comment:
Can you keep a non-alpha version of CpuReducer::sum() ? Always multiplying by alpha would be a big waste of CPU when alpha=1 (at least it's used by the non-compression case), right?

Reply:
ok.